crates/runtime/src/memory.rs

```diff
@@ -5,196 +5,167 @@
 use crate::mmap::Mmap;
 use crate::vmcontext::VMMemoryDefinition;
 use crate::ResourceLimiter;
-use anyhow::{bail, Result};
+use anyhow::{bail, format_err, Result};
 use more_asserts::{assert_ge, assert_le};
 use std::convert::TryFrom;
-use wasmtime_environ::{MemoryPlan, MemoryStyle, WASM_MAX_PAGES, WASM_PAGE_SIZE};
+use wasmtime_environ::{MemoryPlan, MemoryStyle, WASM32_MAX_PAGES, WASM64_MAX_PAGES};
+
+const WASM_PAGE_SIZE: usize = wasmtime_environ::WASM_PAGE_SIZE as usize;
+const WASM_PAGE_SIZE_U64: u64 = wasmtime_environ::WASM_PAGE_SIZE as u64;
 
 /// A memory allocator
 pub trait RuntimeMemoryCreator: Send + Sync {
     /// Create new RuntimeLinearMemory
-    fn new_memory(&self, plan: &MemoryPlan) -> Result<Box<dyn RuntimeLinearMemory>>;
+    fn new_memory(
+        &self,
+        plan: &MemoryPlan,
+        minimum: usize,
+        maximum: Option<usize>,
+    ) -> Result<Box<dyn RuntimeLinearMemory>>;
 }
 
 /// A default memory allocator used by Wasmtime
 pub struct DefaultMemoryCreator;
 
 impl RuntimeMemoryCreator for DefaultMemoryCreator {
     /// Create new MmapMemory
-    fn new_memory(&self, plan: &MemoryPlan) -> Result<Box<dyn RuntimeLinearMemory>> {
-        Ok(Box::new(MmapMemory::new(plan)?) as _)
+    fn new_memory(
+        &self,
+        plan: &MemoryPlan,
+        minimum: usize,
+        maximum: Option<usize>,
+    ) -> Result<Box<dyn RuntimeLinearMemory>> {
+        Ok(Box::new(MmapMemory::new(plan, minimum, maximum)?))
     }
 }
 
 /// A linear memory
 pub trait RuntimeLinearMemory: Send + Sync {
-    /// Returns the number of allocated wasm pages.
-    fn size(&self) -> u32;
+    /// Returns the number of allocated bytes.
+    fn byte_size(&self) -> usize;
 
-    /// Returns the maximum number of pages the memory can grow to.
+    /// Returns the maximum number of bytes the memory can grow to.
     /// Returns `None` if the memory is unbounded.
-    fn maximum(&self) -> Option<u32>;
+    fn maximum_byte_size(&self) -> Option<usize>;
 
-    /// Grow memory by the specified amount of wasm pages.
+    /// Grow memory to the specified amount of bytes.
     ///
     /// Returns `None` if memory can't be grown by the specified amount
     /// of wasm pages.
```

Comment:
I think this commit is now out of date as it's now taking a size rather than page delta?